src/panels/lovelace/cards/energy/common/energy-chart-options.ts

```diff
@@ -242,3 +254,21 @@ export function fillDataGapsAndRoundCaps(datasets: BarSeriesOption[]) {
     }
   });
 }
+
+export function getCompareTransform(start: Date, compareStart?: Date) {
+  if (!compareStart) {
+    return (ts: Date) => ts;
+  }
+  const compareYearDiff = differenceInYears(start, compareStart);
+  const compareIsYear =
+    compareYearDiff !== 0 && start.getTime() === startOfYear(start).getTime();
```

Comment:
This only tells us the start is the start of the year, but the period could span multiple years right? Then all of those years would be set to the same year?

Reply:
This was only handling the common case of year comparison. Updated to handle more years or months now. Description updated as well

Reply:
Yeah, noticed your changes! Looks good :-)